src/transforms/remap.rs

```diff
@@ -650,6 +659,72 @@ fn push_dropped(
     output.push_named(DROPPED, event)
 }
 
+/// If the VRL returns a value that is not an array (see [`merge_array_definitions`]),
+/// or an object, that data is moved into the `message` field.
+fn move_field_definitions_into_message(mut definition: schema::Definition) -> schema::Definition {
+    let mut message = Kind::never();
```

Comment:
Why not copy the entire `definition.event_kind_mut()`, and remove `object` and `array`, rather than try to move everything else over? If a new `Kind` variant is added, this can break pretty easily.